tests/potential/test_pair.py

```diff
@@ -97,6 +97,150 @@ def test_derivative(self):
+
+        self.assertEqual(s.coeff['1','1']['r-0'].value, 1)
+        self.assertEqual(s.coeff['1','1']['r-0'].const, True)
+        self.assertEqual(s.coeff['1','1']['r-1'].value, 2)
+        self.assertEqual(s.coeff['1','1']['r-1'].const, True)
+        self.assertEqual(s.coeff['1','1']['r-2'].value, 3)
+        self.assertEqual(s.coeff['1','1']['r-2'].const, True)
+
+        self.assertEqual(s.coeff['1','1']['knot-0'].value, 5)
+        self.assertEqual(s.coeff['1','1']['knot-0'].const, False)
+        self.assertEqual(s.coeff['1','1']['knot-1'].value, 3)
+        self.assertEqual(s.coeff['1','1']['knot-1'].const, False)
+        self.assertEqual(s.coeff['1','1']['knot-2'].value, 1)
+        self.assertEqual(s.coeff['1','1']['knot-2'].const, True)
+
+        #test value mode
+        s = pair.Spline(types=('1',), num_knots=3, mode='value')
+        s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
+
+        self.assertEqual(s.coeff['1','1']['r-0'].value, 1)
+        self.assertEqual(s.coeff['1','1']['r-0'].const, True)
+        self.assertEqual(s.coeff['1','1']['r-1'].value, 2)
+        self.assertEqual(s.coeff['1','1']['r-1'].const, True)
+        self.assertEqual(s.coeff['1','1']['r-2'].value, 3)
+        self.assertEqual(s.coeff['1','1']['r-2'].const, True)
+
+        self.assertEqual(s.coeff['1','1']['knot-0'].value, 9)
+        self.assertEqual(s.coeff['1','1']['knot-0'].const, False)
+        self.assertEqual(s.coeff['1','1']['knot-1'].value, 4)
+        self.assertEqual(s.coeff['1','1']['knot-1'].const, False)
+        self.assertEqual(s.coeff['1','1']['knot-2'].value, 1)
+        self.assertEqual(s.coeff['1','1']['knot-2'].const, True)
+
+        #test invalid r and u shapes
+        r_arr = [2,3]
+        with self.assertRaises(ValueError):
+            s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
+
+        r_arr = [1,2,3]
+        u_arr = [1,2]
+        with self.assertRaises(ValueError):
+            s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
+
+    def test_energy(self):
+        """Test _energy method"""
+        r_arr = [1,2,3]
+        u_arr = [9,4,1]
+
+        #test diff mode
+        s = pair.Spline(types=('1',), num_knots=3)
+        s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
+        u_actual = np.array([6.25,2.25,1])
+        u = s._energy(pair=('1','1'), r=[1.5,2.5,3.5])
+        np.testing.assert_allclose(u, u_actual)
+
+        #test value mode
+        s = pair.Spline(types=('1',), num_knots=3, mode='value')
+        s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
+        u_actual = np.array([6.25,2.25,1])
+        u = s._energy(pair=('1','1'), r=[1.5,2.5,3.5])
+        np.testing.assert_allclose(u, u_actual)
+
+    def test_force(self):
+        """Test _force method"""
+        r_arr = [1,2,3]
+        u_arr = [9,4,1]
+
+        #test diff mode
+        s = pair.Spline(types=('1',), num_knots=3)
+        s.from_array(pair=('1','1'), r=r_arr, u=u_arr)
+        f_actual = np.array([5,3,0])
```

Comment:
This might be a silly question, but how did you determine what these forces should be? For 3 points, does the interpolator just do linear interpolation?